includes/preflight.inc

```diff
@@ -441,7 +441,7 @@ function _drush_preflight_columns() {
   if (!($columns = getenv('COLUMNS') ?: 0)) {
     // Trying to export the columns using stty.
     exec('stty size 2>&1', $columns_output, $columns_status);
-    if (!$columns_status) $columns = preg_replace('/\d+\s(\d+)/', '$1', $columns_output[0], -1, $columns_count);
+    if (!$columns_status && is_numeric($columns_output[0])) $columns = preg_replace('/\d+\s(\d+)/', '$1', $columns_output[0], -1, $columns_count);
```

Comment:
```suggestion
    $matched = FALSE;
    if (!$columns_status) {
        preg_match('/\d+\s(\d+)/', $columns_output[0], $matches, 0);
        if (isset($matches[1])) {
            $columns = $matches[1];
            $matched = TRUE;
        }
    }
```

Reply:
Looks good in concept. Maybe `$matched = preg_match...` would be sufficient if you made the regex bounded with `^` and `$` for an exact match.

Reply:
```suggestion
    $matched = FALSE;
    if (!$columns_status && $matched = preg_match('/^\d+\s(\d+)$/', $columns_output[0], $matches, 0)) {
        $columns = $matches[1];
    }
```
Good idea, it could be improved like this.